tokio-test/src/macros.rs

```diff
@@ -0,0 +1,41 @@
+//! A collection of useful macros for testing futures and tokio based code
+
+/// Assert if a future is ready
+#[macro_export]
+macro_rules! assert_ready {
+    ($e:expr) => {{
+        match $e {
+            Ok(::futures::Async::Ready(v)) => v,
+            Ok(_) => panic!("not ready"),
+            Err(e) => panic!("error = {:?}", e),
+        }
+    }};
+}
+
+/// Asset if the Future is not ready
+#[macro_export]
+macro_rules! assert_not_ready {
+    ($e:expr) => {{
+        match $e {
+            Ok(::futures::Async::NotReady) => {}
+            Ok(::futures::Async::Ready(v)) => panic!("ready; value = {:?}", v),
```

Comment:
Again, would be nice to stringify the expression we're asserting is `NotReady` here..